### DISCUSSION TITLE
osu! not using files in ~/.local/share/osu/

### DISCUSSION THREAD

**Discussion by StalkyBoi43815** [ppy/osu#30800] [Q&A] [answered]
Basically title, I can't access any of my beatmaps or skins. I have no idea where osu! is importing new stuff. Tried running it in the terminal but it shows nothing.
[logs.zip](https://github.com/user-attachments/files/17834911/logs.zip)

**Comment by Stoppedpuma** [comment]
According to your logs, you seem to be running an old version. Running the latest version should fix this.

**Reply by StalkyBoi43815** [reply]
But I'm using the latest AppImage. Don't know where the AppImage's logs are, I just ran both that and the old .desktop. ~/.local/share/osu/ is the old version's dir. That's what my problem is. I don't know where the latest AppImage is saving stuff, these are all the logs I could find. I literally `sudo find / -type d -name` that bitch. Could find nothing else, but the maps I import somehow stay.

**Reply by Stoppedpuma** [reply]
The latest logs show you ran an October build. Likely the issue here is you're running an old AppImage. Try downloading and running the latest one.

> sudo find / -type d -name

Files are named by hash, this is likely why you're not finding anything.

**Reply by StalkyBoi43815** [reply]
I looked for folders named "osu" not hashes. Also just ran the latest AppImage and no change. Logs after running this: 
[logs.zip](https://github.com/user-attachments/files/17835194/logs.zip)

**Reply by Stoppedpuma** [reply]
Okay, you seem to be running the latest version now! I would check in `~/.local/share/osu` for a file named realm backup or something around those lines and replace the existing one (after backing it up), it should be bigger in file size then the existing client.realm file. If that doesn't work, maybe see if updating your custom rulesets fixes this?

**Reply by StalkyBoi43815** [reply]
Didn't work, removed all rulesets instead of updating cuz who cares lol, still nothing. I even deleted all **files** in `~/.local/share/osu` but left the folders, still nothing.

**Reply by Stoppedpuma** [reply, marked as answer]
> I even deleted all files in ~/.local/share/osu but left the folders, still nothing.

Well now those files are "lost" without the realm database. Not much I can do to help here? Maybe someone else knows if there's a way to rebuild the database but I'm pretty sure peppy has said in the past that not much can be done? I would see if you can retrieve the files from your trash bin, otherwise file recovery if they have been deleted. If you're unable to get the database back then I would assume you'll have to redownload everything again?

**Reply by StalkyBoi43815** [reply]
Fucking hell mate, those database files are unreadable anyways. I've backed up my osu! without them many times, it still survived. About to check the trash, but what the fuck is this anyways? This only started happening after the last update. Man, those guys that maintain the AUR package need to be fired, they aren't doing shit.

**Reply by Stoppedpuma** [reply]
> but what the fuck is this anyways

See here for more details: https://github.com/ppy/osu/wiki/User-file-storage

> those guys that maintain the AUR package need to be fired, they aren't doing shit.

That's a community maintained package, not much can be said besides stick to the official AppImage to lower chances of random breakage?

**Reply by StalkyBoi43815** [reply]
Went through my trash and restored all of them one by one, and finally found the right one. Finally recovered over 6000 beatmaps (yeah ik it aint much but its honest work)
![enshot 2024-11-20](https://github.com/user-attachments/assets/a361232c-1703-4c7d-a138-b99adde60d7f) All that over this update goddammit like barely any changes as usually? But then again I know nothing about game dev't. Also excuse the default settings, wiped that too. And thanks for the really neat idea, looking through the trash worked.

**Reply by Stoppedpuma** [reply]
Glad you could recover your files! Be sure to make backups in the future before dangerous actions!

**Reply by bdach** [reply]
> not much can be said besides stick to the official AppImage to lower chances of random breakage?

as if we have a much better track record with the appimage (see: updater fiasco from latest release). probably "stop using linux" would be the advice here

> All that over this update goddammit like barely any changes as usually?

[so this is barely any changes](https://osu.ppy.sh/home/changelog/lazer/2024.1115.1), noted.

**Reply by StalkyBoi43815** [reply]
I didn't mean like barely any changes in that sense, you know? Like, not many instantly noticeable changes and stuff. Like, I only noticed the editor stuff not dimming, the reduced transparency (screenshot) and the drag thingy focus. I'm not saying the people working on the game are doing a shit job. Yall are doing god's work tbh.

**Reply by peppy** [reply]
If you have a change you'd like to see, please upvote the discussions or issues.
